### ISSUE TITLE
New check: JavadocBlockStyle

### ISSUE THREAD

**Issue by mohitsatr** [checkstyle/checkstyle#17832]
From:  [7.1.1 General Form](https://google.github.io/styleguide/javaguide.html#s7.1.1-javadoc-multi-line) and https://github.com/google/google-java-format/issues/1279#issuecomment-3320778635

This check ensures:
1. Leading asterisk is always present.
2. Preferred style is to put /** and */ on their own lines (unless the entire javadoc comment fits on a single line)

Expected config:
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocBlockStyle"/>
    </module>
</module>
```

Some other examples showing how the `JavadocBlockStyle` should work:

1. Missing leading asterisk

```java
public class JavadocAsteriskFormat {

    /**
     * Leading Asterisk is missing in next line.
        it should not be allowed   // violation
     */
    void testMethod() {}

    /*
     * Some Method.
     * 
       @param num some-number.
       @param path some-path.   // violation
     */
    void testMethod2(int num, String path) {}

    /**
     * Some method.
     *
      @return randomString.   // violation
     */
    String testMethod3() {
        return "method";
    }
}
```

Expected Output:
```
$ java -jar checkstyle-11.0.0-all.jar -c config.xml JavadocAsteriskFormat.java 
Starting audit...
[WARN] /mnt/5D92528E6B945467/test/testing/JavadocAsteriskFormat.java:5: Leading asterisk is missing. [JavadocBlockStyle]
[WARN] /mnt/5D92528E6B945467/test/testing/JavadocAsteriskFormat.java:12: Leading asterisk is missing. [JavadocBlockStyle]
[WARN] /mnt/5D92528E6B945467/test/testing/JavadocAsteriskFormat.java:13: Leading asterisk is missing. [JavadocBlockStyle]
[WARN] /mnt/5D92528E6B945467/test/testing/JavadocAsteriskFormat.java:20: Leading asterisk is missing. [JavadocBlockStyle]
Audit done.
```

2. `/**` and `*/` not on their own lines

```java
public class JavadocAsteriskFormat {

    /**
     * Javadoc closing asterisk not on it's own line. */  // violation
    private void testMethod1() {}

    /** Javadoc opening asterisk not on it's own line. so this    // violation
     * is a invalid code.
     */
    private void testMethod2() {}

    /**
     * Returns a random number.
     * 
     * @return random number */   // violation
    private int testMethod3() {
        return 1;
    }
}
```

Expected output:
```
$ java -jar checkstyle-11.0.0-all.jar -c config.xml JavadocAsteriskFormat.java 
Starting audit...
[WARN] /mnt/5D92528E6B945467/test/testing/JavadocAsteriskFormat.java:4: Javadoc closing asterisk must be on it's own line. [JavadocBlockStyle]
[WARN] /mnt/5D92528E6B945467/test/testing/JavadocAsteriskFormat.java:7: Javadoc opening asterisks must be on it's own line. [JavadocBlockStyle]
[WARN] /mnt/5D92528E6B945467/test/testing/JavadocAsteriskFormat.java:15: Javadoc closing asterisk must be on it's own line. [JavadocBlockStyle]
Audit done.
```

3. Entire Javadoc comment on a single line

```java
/** This is main class */
public class JavadocAsteriskFormat {
```

Expected output:
```
$ java -jar checkstyle-11.0.0-all.jar -c config.xml JavadocAsteriskFormat.java 
Starting audit...
Audit done.
```

**Comment by romani** [comment]
>Leading asterisk is always present.

https://checkstyle.org/checks/javadoc/javadocmissingleadingasterisk.html#JavadocMissingLeadingAsterisk

Will be done at https://github.com/checkstyle/checkstyle/issues/17696

>The single-line form may be substituted when the entirety of the Javadoc block (including comment markers) can fit on a single line

So it is not a strict rule, so both is ok.


--------

I don't think we need anything to add.
